ui/src/app/app-routing.module.ts

```diff
@@ -32,233 +31,242 @@ import {environment} from 'environments/environment';
-                  data: {
-                    title: 'Concept Set',
-                    breadcrumb: {
-                      value: 'Param: Concept Set Name',
-                    }
-                  },
-                  resolve: {
-                    conceptSet: ConceptSetResolver,
-                  }
-                }]
-              }]
+                resolve: {
+                  cohort: CohortResolver,
+                }
+              }
+            ]
+          },
+          {
+            path: 'concepts',
+            component: ConceptHomepageComponent,
+            data: {
+              title: 'Search Concepts',
+              breadcrumb: {
+                value: 'Concepts',
+                intermediate: true
+              }
+            }
+          },
+          {
+            path: 'concepts/sets',
+            data: {
+              breadcrumb: {
+                value: 'Concept Sets',
+                intermediate: true
+              }
+            },
+            children: [{
+              path: '',
+              component: ConceptSetListComponent,
+              data: {
+                title: 'View Concept Sets',
+              }
+            }, {
+              path: ':csid',
+              component: ConceptSetDetailsComponent,
+              data: {
+                title: 'Concept Set',
+                breadcrumb: {
+                  value: 'Param: Concept Set Name',
+                }
+              },
+              resolve: {
+                conceptSet: ConceptSetResolver,
+              }
+            }]
           }]
-      },
-      {
-        path: 'admin/review-workspace',
-        component: AdminReviewWorkspaceComponent,
-        data: {title: 'Review Workspaces'}
-      }, {
-        path: 'admin/review-id-verification',
-        component: AdminReviewIdVerificationComponent,
-        data: {title: 'Review ID Verifications'}
-      }, {
-        path: 'profile',
-        component: ProfilePageComponent,
-        data: {title: 'Profile'}
-      }, {
-        path: 'settings',
-        component: SettingsComponent,
-        data: {title: 'Settings'}
-      }, {
-        path: 'workspaces/build',
-        component: WorkspaceEditComponent,
-        data: {title: 'Create Workspace', mode: WorkspaceEditMode.Create}
-      }
-    ]
+      }]
+  },
+  {
+    path: 'admin/review-workspace',
+    component: AdminReviewWorkspaceComponent,
+    data: {title: 'Review Workspaces'}
+  }, {
+    path: 'admin/review-id-verification',
+    component: AdminReviewIdVerificationComponent,
+    data: {title: 'Review ID Verifications'}
+  }, {
+    path: 'profile',
+    component: ProfilePageComponent,
+    data: {title: 'Profile'}
+  }, {
+    path: 'settings',
+    component: SettingsComponent,
+    data: {title: 'Settings'}
+  }, {
+    path: 'workspaces/build',
+    component: WorkspaceEditComponent,
+    data: {title: 'Create Workspace', mode: WorkspaceEditMode.Create}
   }
 ];
 
+fp.forEach(route => {
```

Comment:
nit: is it really preferable to use `fp` for this? Why not just use built-ins `[].forEach()`

Reply:
No difference. Changing.